packages/vhd-lib/merge.js

```diff
@@ -30,19 +31,13 @@ function makeThrottledWriter(handler, path, delay) {
 // Merge vhd child into vhd parent.
 //
 // TODO: rename the VHD file during the merge
-module.exports = limitConcurrency(2)(async function merge(
-  parentHandler,
-  parentPath,
-  childHandler,
-  childPath,
-  { onProgress = noop } = {}
-) {
+module.exports = limitConcurrency(2)(async function merge(handler, parentPath, childPath, { onProgress = noop } = {}) {
```

Comment:
We misunderstood each other this morning, I believe we should pass `child` as a `Vhd` here and let the calling code instantiate it.

Reply:
it seems quite strange to pass a path for the parent and a  vhd for the child. `mergeVhd` should be the one openning the parent since it's the only one that know if we should read the secondary footer or no

Reply:
It's more powerful to have a `Vhd` for the child because it allows to creates it the way we need, either be it a `VhdSynthetic`, a VHD created directly in memory or opened from another remote handler.

I agree it makes sense to pass the parent as path and it's a bit clunky to have the child and the parent be passed in a different way, but I believe we can make it clear enough in the param names (e.g. `parentPath` and `childVhd`).

What do you think?

Reply:
I think it's not needed and won't bring enough benefict to mandate a refactoring . If we want to keep the possibility of merging from one remote to another, I would prefer to roll back my changes with the  childHandler

Reply:
After discussing this, we decided to go the simplest way: leave the `parentHandler` and `childHandler` params and let `mergeVhd` open the `VhdSynthetic` if `childPath` is an array.